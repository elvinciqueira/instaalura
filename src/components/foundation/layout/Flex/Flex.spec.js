import {Flex} from './'

describe('<Flex />', () => {
  test('renders', () => {
    const flex = rendererCreateWithTheme(<Flex />).toJSON()
    expect(flex).toMatchSnapshot()
  })

  test('alignItems props set align-items', () => {
    const flex = rendererCreateWithTheme(<Flex alignItems="center" />).toJSON()
    expect(flex).toMatchSnapshot()
    expect(flex).toHaveStyleRule('align-items', 'center')
  })

  test('justifyContent props set justify-content', () => {
    const flex = rendererCreateWithTheme(
      <Flex justifyContent="center" />,
    ).toJSON()
    expect(flex).toMatchSnapshot()
    expect(flex).toHaveStyleRule('justify-content', 'center')
  })

  test('flexWrap props set flex-wrap', () => {
    const flex = rendererCreateWithTheme(<Flex flexWrap="wrap" />).toJSON()
    expect(flex).toMatchSnapshot()
    expect(flex).toHaveStyleRule('flex-wrap', 'wrap')
  })
})
