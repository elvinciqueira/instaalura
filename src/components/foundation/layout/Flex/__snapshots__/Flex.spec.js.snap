// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Flex /> alignItems props set align-items 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0 c1"
/>
`;

exports[`<Flex /> flexWrap props set flex-wrap 1`] = `
.c0 {
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0 c1"
/>
`;

exports[`<Flex /> justifyContent props set justify-content 1`] = `
.c0 {
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0 c1"
/>
`;

exports[`<Flex /> renders 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

<div
  className="c0"
/>
`;
